### DISCUSSION TITLE
Migrate to 1.0.0-f899db8: How to get cmock to run

### DISCUSSION THREAD

**Discussion by CJCombrink** [ThrowTheSwitch/Ceedling#956] [Q&A]
Hi, 

I am trying to migrate an existing ceedling project to version 1.0.0. 
I have made some progress, but it seems like cmock is not running.

Is there some docs on what is needed that I am missing that contains information on how to migrate or enable cmock for the new project file?

The symptom is that when running `ceedling  test:all` I get a ` fatal error: mock_platform_mocking.h: No such file or directory` compiler error. I have searched the file system and does not find one, thus I am assuming cmock was never executed.

I have tried `:use_test_preprocessor: :mocks` and `:use_test_preprocessor: :all`  but still can't get cmock to run.

_PS: I am very new to ceedling, cmock, etc. I am a new member of a team using these tools_ 

Ceedling 0.31.1 definitely generated this header and associated other files. 

<details>
<summary>project.yml</summary>

```
:project:
  :compile_threads: :auto
  :test_threads: :auto
  :use_test_preprocessor: :all
  :use_backtrace: :simple
  :build_root: build

:paths:
  :test:
    - +:test/**
    - -:test/support
  :include:
    - lib/**
    - source/**
    - test/support
  :source:
    - +:source/**
    - +:lib/**

:defines:
  :common: &common_defines []
  :test:
    - *common_defines
    - TEST
    - OS_FREERTOS
    - SIMULATION
    - __ARM_ARCH_6M__
    - ALGORITHM_EXPECTED_SAMPLING_RATE=100
  :test_preprocess:
    - *common_defines
    - TEST

:flags:
  :test:
    :compile:
      - -std=c99 
  :release: 
    :compile: 
      - -std=c99

:cmock:
  :mock_prefix: mock_
  :when_no_prototypes: :warn
  :enforce_strict_ordering: TRUE
  :treat_externs: :include
  :plugins:
    - :ignore
    - :callback
    - :return_thru_ptr
    - :ignore_arg
    - :expect_any_args
    - :array
  :treat_as:
    uint8:    HEX8
    uint16:   HEX16
    uint32:   UINT32
    int8:     INT8
    bool:     UINT8

```
</details>

**Comment by mvandervoord** [comment]
If you type `ceedling files:header` does it list `platform_mocking.h`?

**Reply by CJCombrink** [reply]
Yes it does
```
find . -type f -iname "platform_mocking.h"
./test/support/platform_mocking.h
```
and
```
ceedling files:header | grep platform_mocking.h
 - test/support/platform_mocking.h
 ```

**Comment by JuPrgn** [comment]
I had something similar when I upgraded my project before running `ceedling clobber` and removing all 0.31.1  artifacts in the Ceedling build folder

**Reply by CJCombrink** [reply]
Thanks, this however is a clean install (docker) and I use `git clean -fdx` rather often to ensure it is a clean repo

**Comment by CJCombrink** [comment]
It appears that I missed the following under `:projects`

```
:project:
  ...
  :use_mocks: TRUE
```
